### ISSUE TITLE
FP single-component-path-imports :   self::<import>::...

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#10549]
### Summary

.

### Lint Name

single_component_path_imports

### Reproducer

I tried this code:

```rust
use std;

fn main() {
    let _ = self::std::io::stdout();
}
```

I saw this happen:

```
warning: this import is redundant
 --> src/main.rs:1:1
  |
1 | use std;
  | ^^^^^^^^ help: remove it entirely
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
  = note: `#[warn(clippy::single_component_path_imports)]` on by default
```

Applying the suggestion breaks the build:
````
The following errors were reported:
error[E0433]: failed to resolve: could not find `std` in the crate root
 --> src/main.rs:8:19
  |
8 |     let _ = self::std::io::stdout();
  |                   ^^^ could not find `std` in the crate root

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
Original diagnostics will follow.
````

### Version

```text
rustc 1.70.0-nightly (8be3c2bda 2023-03-24)
binary: rustc
commit-hash: 8be3c2bda6b683f87b24714ba595e8b04faef54c
commit-date: 2023-03-24
host: x86_64-unknown-linux-gnu
release: 1.70.0-nightly
LLVM version: 15.0.7
```
